feat(Input): support pattern (#3091)
TuSimple/naive-ui: src/input/demos/enUS/index.demo-entry.md

| show-password-on | `'click' \| 'mousedown'` | `undefined` | The event to show the password. |  |
| size | `'small' \| 'medium' \| 'large'` | `'medium'` | Input size. |  |
| status | `'success' \| 'warning' \| 'error'` | `undefined` | Validaiton status. | 2.25.0 |

> | pattern | `(value: string) => void` | `undefined` | Check the current entry, if return the `False`, input will not respond this entry. | NEXT_VERSION |

Reviewer: ```suggestion
| pattern | `(value: string) => boolean` | `undefined` | Check the incoming value, if it returns `false`, input will not be accepted. | NEXT_VERSION |
```
